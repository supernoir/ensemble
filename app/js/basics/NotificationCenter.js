import React from 'react';
import Notification from './Notification';
import PropTypes from 'prop-types';

const MESSAGES_LIMIT = 5;

/**
 * Class NotificationCenter
 * Wrapping component to handle a set of notifications
 */
export default class NotificationCenter extends React.Component {
	renderNotifications = messages => {
		return (
			<div className="app_notification-container">
				{messages.map((message, index) => {
					return <Notification key={`${message.type}-${index}`} type={message.type} content={message.content} description={message.description} />;
				})}
			</div>
		);
	};

	render() {
		if (this.props.messages !== void 0) {
			if (this.props.messages.length < MESSAGES_LIMIT) {
				return this.renderNotifications(this.props.messages);
			} else {
				this.props.messages.shift();
				return this.renderNotifications(this.props.messages);
			}
		} else {
			return null;
		}
	}
}

NotificationCenter.propTypes = {
	messages: PropTypes.shape({
		type       : PropTypes.string,
		content    : PropTypes.string,
		description: PropTypes.string
	})
};