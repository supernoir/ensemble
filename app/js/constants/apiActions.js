const API_ACTIONS = {
	GET   : 'get',
	POST  : 'post',
	PUT   : 'put',
	DELETE: 'delete'
};


export default API_ACTIONS;