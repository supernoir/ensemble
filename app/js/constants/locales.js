/**
 * Available Locales Constant
 */
const LOCALES = [
	{ locale: 'en-US', code: 'us', name: 'lang.en_US' },
	{ locale: 'es-ES', code: 'es', name: 'lang.es_ES' },
	{ locale: 'de-DE', code: 'de', name: 'lang.de_DE' }
];

export default LOCALES;