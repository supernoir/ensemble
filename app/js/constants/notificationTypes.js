/**
 * Notification Types Constant
 */
export const notificationTypes = {
	INFO   : 'info',
	WARN   : 'warning',
	ERROR  : 'error',
	SUCCESS: 'success'
};
