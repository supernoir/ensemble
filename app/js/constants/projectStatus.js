/**
 * Project Status Constant
 */
export const projectStatus = {
	DRAFT      : 'draft',
	IN_PROGRESS: 'inprogress',
	IN_REVIEW  : 'inreview',
	EDITING    : 'editing',
	PUBLISHED  : 'published'
};

export const projectStatusColormapping = {
	draft      	: 'grey',
	inprogress: 'blue',
	inreview  	: 'red',
	editing    	: 'yellow',
	published  	: 'green'
};