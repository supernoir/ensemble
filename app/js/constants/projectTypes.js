export const projectTypes = {
	SHORTSTORY: 'shortstory',
	SCRIPT    : 'script',
	NOVEL     : 'novel',
	ESSAY     : 'essay',
	poem      : 'poem'
};
