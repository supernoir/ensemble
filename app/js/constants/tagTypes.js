export const tagTypes = {
	// Specific tag for only one project
	PROJECT      : 'project',
	// User-based tag
	USER         : 'user',
	// Global tag for all users
	GLOBAL       : 'global',
	// tag represents project state
	PROJECT_STATE: 'project_state',
	// tag represents project type
	PROJECT_TYPE : 'project_type'
};
