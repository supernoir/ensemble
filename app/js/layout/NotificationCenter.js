import React from 'react';
import Notification from './Notification';

const MESSAGES_LIMIT = 5;

export default class NotificationCenter extends React.Component {
	renderNotifications = messages => {
		return (
			<div className="app_notification-container">
				{messages.map(message => {
					return <Notification type={message.type} content={message.content} description={message.description} />;
				})}
			</div>
		);
	};

	render() {
		if (this.props.messages !== void 0) {
			if (this.props.messages.length < MESSAGES_LIMIT) {
				return this.renderNotifications(this.props.messages);
			} else {
				this.props.messages.shift();
				return this.renderNotifications(this.props.messages);
			}
		} else {
			return null;
		}
	}
}
