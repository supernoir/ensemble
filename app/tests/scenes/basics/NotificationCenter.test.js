import React from 'react';
import { shallow, mount, render } from 'enzyme';
import { notificationTypes } from '../../../js/constants/notificationTypes';
import { Provider } from 'react-redux';

const testData = {
	messages: [
		{
			type       : notificationTypes.ERROR,
			content    : 'A test error occurred',
			description: 'Something went wrong'
		},
		{
			type       : notificationTypes.WARN,
			content    : 'A test warning occurred',
			description: 'Something went wrong'
		},
		{
			type       : notificationTypes.INFO,
			content: 'Here be dragons',
		},
		{
			type       : notificationTypes.SUCCESS,
			content    : 'Yay you did it!',
			description: 'A job well done'
		},
	],
	store: {
		dispatch : jest.fn(),
		getState : jest.fn(),
		subscribe: jest.fn()
	}
};

// Import Components
import NotificationCenter from '../../../js/containers/MessengerContainer';

test('NotificationCenter -- Snapshot Test', () => {
	const wrapper = shallow(<Provider store={testData.store}>
		<NotificationCenter messages={testData.messages} />
	</Provider>
	);
	expect(wrapper).toMatchSnapshot();
});

/* test('NotificationCenter -- Type Test', () => {
	const wrapper = render(<Provider store={testData.store}>
		<NotificationCenter messages={testData.messages} />
	</Provider>
	);
	expect(wrapper.find(NotificationCenter).children()).toHaveLength(4);
}); */
