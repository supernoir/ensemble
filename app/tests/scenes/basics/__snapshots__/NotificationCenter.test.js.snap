// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationCenter -- Snapshot Test 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [MockFunction],
        "getState": [MockFunction] {
          "calls": Array [
            Array [],
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "subscribe": [MockFunction] {
          "calls": Array [
            Array [
              [Function],
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [MockFunction],
          "getState": [MockFunction] {
            "calls": Array [
              Array [],
              Array [],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          },
          "subscribe": [MockFunction] {
            "calls": Array [
              Array [
                [Function],
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          },
        },
        "unsubscribe": undefined,
      },
    }
  }
>
  <Connect(NotificationCenter)
    messages={
      Array [
        Object {
          "content": "A test error occurred",
          "description": "Something went wrong",
          "type": "error",
        },
        Object {
          "content": "A test warning occurred",
          "description": "Something went wrong",
          "type": "warning",
        },
        Object {
          "content": "Here be dragons",
          "type": "info",
        },
        Object {
          "content": "Yay you did it!",
          "description": "A job well done",
          "type": "success",
        },
      ]
    }
  />
</ContextProvider>
`;
